Add more places for Donation sections. Refactored existing logic.

digitaldirk opened the pull request:
**Added optional donation sections to:**
- Footer
- Introduction
- New Support me page in Nav Menu (with new description field)

Made donation under blog post optional. Refactored/moved existing code. Discussion at: #357 

**BREAKING CHANGES**
Moved existing support configuration into new `SupportMeConfiguration` out from `ApplicationConfiguration`

**Preview:**
Footer
![supportme1](https://github.com/user-attachments/assets/479c7e62-a2c5-4192-bfbf-768c9e0d1a9d)
Introduction
![supportme2](https://github.com/user-attachments/assets/5dd0f06c-d533-45eb-bf64-903c1492c75e)
Support me page
![supportme3](https://github.com/user-attachments/assets/5b297f7d-3cac-4d7a-a931-5a65c6e1786a)

**Still Needed**
@linkdotnet (or anyone else), please make the following changes to complete this PR.
1. Support me page in nav menu needs an icon (placeholder used).
2. New support me page needs css formatting (centering, padding, font size?,...)

**How to configure new support me settings:**
```json
"SupportMe": {
	"KofiToken": "ABC123",
	"GithubSponsorName": "your-tag-here",
	"PatreonName": "your-tag-here",
	"ShowUnderBlogPost": true,
	"ShowUnderIntroduction": true,
	"ShowInFooter": true,
	"ShowSupportMePage": true,
	"SupportMePageDescription": "Buy my book here: [My Blazor Book](https://google.com) or please contribute to my open-source project here: [My Awesome Repo](https://github.com) . This can be **markdown**."
}
```

linkdotnet commented on src/LinkDotNet.Blog.Web/Features/SupportMe/SupportMePage.razor:
```diff
@@ -0,0 +1,26 @@
+@using LinkDotNet.Blog.Web.Features.SupportMe.Components
+@inject IOptions<ApplicationConfiguration> AppConfiguration
+@inject IOptions<SupportMeConfiguration> SupportConfiguration
+@inject AuthenticationStateProvider AuthenticationStateProvider
```
Seems unused

linkdotnet commented on src/LinkDotNet.Blog.Web/Features/SupportMe/SupportMePage.razor:
```diff
@@ -0,0 +1,26 @@
+@using LinkDotNet.Blog.Web.Features.AboutMe.Components
+@using LinkDotNet.Blog.Domain
+@using LinkDotNet.Blog.Web.Features.SupportMe.Components
+@inject IOptions<ApplicationConfiguration> AppConfiguration
```
Seems unused

linkdotnet commented on src/LinkDotNet.Blog.Web/Features/ShowBlogPost/ShowBlogPostPage.razor:
```diff
@@ -12,6 +13,7 @@
 @inject IUserRecordService UserRecordService
 @inject IOptions<ApplicationConfiguration> AppConfiguration
```
Probably also here not used anymore

digitaldirk replied:
Still used

linkdotnet commented on src/LinkDotNet.Blog.Web/Features/ShowBlogPost/ShowBlogPostPage.razor:
```diff
@@ -81,8 +83,11 @@ else if (BlogPost is not null)
-			@if (AppConfiguration.Value.ShowSimilarPosts)
+            @if (SupportConfiguration.Value.ShowUnderBlogPost)
+            {
+                <LinkDotNet.Blog.Web.Features.SupportMe.Components.DonationSection />   
```
nitpick:You could import the namespace `LinkDotNet.Blog.Web.Features.SupportMe.Components` so that the tag name is only: `<DonationSection>`

digitaldirk replied:
Fixed thanks

linkdotnet reviewed (commented):
really cool work.
The icon I can do after we merged. There is a readme.md inside the icons, but not sure how easy that is you :D 

1. Some tests are failing at the moment because of the new structure
2. A small entry inside the `MIGRATION.md` would be nice that describes what has to be done for folks that are migrating.
3. If you like and have time a small part inside the documentation (markdown files under `docs/`).

Thanks for the initiative @digitaldirk

digitaldirk commented:
Will work on reviews late tomorrow for me

linkdotnet commented:
Thanks! Take your time and let me know if I can help

digitaldirk commented:
99% there I think 😅 will look into `SupportMePageDescription` tomorrow, or feel free if its a quick fix for you.

Added/fixed tests, add/moved documentation, and addressed your comments.

Thank you

linkdotnet commented:
Not sure if I can push directly to your branch - I'll try :D

linkdotnet commented:
Okay - that works. Nice

linkdotnet commented:
Nice work @digitaldirk - much appreciated. I would merge it if there aren't any open points from your side?
I added the migration text and some tests for the new component you created!

linkdotnet commented:
Lol - i accidentally merged :D. Wanted to rebase on the latest `master`. Well let me know if there where some missing points

digitaldirk commented:
> Lol - i accidentally merged :D. Wanted to rebase on the latest `master`. Well let me know if there where some missing points

No worries. Thank you for cleaning up and adding the tests.

Only things left:
- I didn't find an icon I loved for the support me page nav menu, please pick one for it.
- In 'migration.md' I added notes on the bottom and you added some on the top, could these please be combined so its only in one place.

linkdotnet commented:
I added this one:
<img width="477" alt="image" src="https://github.com/user-attachments/assets/4324b542-f70e-409d-ab2d-77c1df61c166">
